shuffle/cli/src/shared.rs

```diff
@@ -137,50 +137,103 @@ impl DevApiClient {
+
+        DevApiClient::check_response(
+            self.client.get(path.as_str()).send().await?,
+            "GET /transactions failed",
+        )
+        .await
     }
 
-    pub async fn post_transactions(&self, txn_bytes: Vec<u8>) -> Result<Response> {
+    pub async fn post_transactions(&self, txn_bytes: Vec<u8>) -> Result<Value> {
         let path = self.network.join("transactions")?;
-        Ok(self
-            .client
-            .post(path.as_str())
-            .header("Content-Type", mime_types::BCS_SIGNED_TRANSACTION)
-            .body(txn_bytes)
-            .send()
-            .await?)
+
+        DevApiClient::check_response(
+            self.client
+                .post(path.as_str())
+                .header("Content-Type", mime_types::BCS_SIGNED_TRANSACTION)
+                .body(txn_bytes)
+                .send()
+                .await?,
+            "POST /transactions failed",
+        )
+        .await
     }
 
-    async fn get_account_resources(&self, address: AccountAddress) -> Result<Response> {
+    async fn get_account_resources(&self, address: AccountAddress) -> Result<Value> {
         let path = self
             .network
             .join(format!("accounts/{}/resources", address.to_hex_literal()).as_str())?;
-        Ok(self.client.get(path.as_str()).send().await?)
+
+        DevApiClient::check_response(
+            self.client.get(path.as_str()).send().await?,
+            "Failed to get account resources with provided address",
+        )
+        .await
     }
 
-    pub async fn get_account_sequence_number(&self, address: AccountAddress) -> Result<u64> {
-        let resp = self.get_account_resources(address).await?;
-        DevApiClient::check_accounts_resources_response_status_code(&resp.status())?;
-        let json: Vec<Value> = serde_json::from_str(resp.text().await?.as_str())?;
-        DevApiClient::parse_json_for_account_seq_num(json)
+    pub async fn get_account_transactions_response(
+        &self,
+        address: AccountAddress,
+        start: u64,
+        limit: u64,
+    ) -> Result<Value> {
+        let path = self
+            .network
+            .join(format!("accounts/{}/transactions", address).as_str())?;
+
+        DevApiClient::check_response(
+            self.client
+                .get(path.as_str())
+                .query(&[("start", start.to_string().as_str())])
+                .query(&[("limit", limit.to_string().as_str())])
+                .send()
+                .await?,
+            "Failed to get account transactions with provided address",
+        )
+        .await
     }
 
-    fn check_accounts_resources_response_status_code(status_code: &StatusCode) -> Result<()> {
-        match status_code == &StatusCode::from_u16(200)? {
+    async fn check_response(resp: Response, failure_message: &str) -> Result<Value> {
+        let status = resp.status();
+        let json: serde_json::Value = serde_json::from_str(resp.text().await?.as_str())?;
```

Comment:
You can get the `json()` directly without having a `text()` intermediary.

```rust
let json = resp.json().await?.
// or if you really want the serde_json type
let json = resp.json::<serde_json::Value>().await?.
```

https://docs.rs/reqwest/0.9.20/reqwest/struct.Response.html#method.json

@bmwill what are your thoughts on this? Does it matter at all?

Reply:
Yeah it would be preferable to use `.json` instead.

btw I'm in the process of creating a reusable rest client similar to what we have with the jsonrpc one so this should be able to be simplified at some point